src/material/datepicker/datepicker-toggle.ts

```diff
@@ -57,7 +57,7 @@ export class MatDatepickerToggle<D> implements AfterContentInit, OnChanges, OnDe
   private _stateChanges = Subscription.EMPTY;
 
   /** Datepicker instance that the button will toggle. */
-  @Input('for') datepicker: MatDatepicker<D>;
+  @Input('for') datepicker: MatDatepickerBase<MatDatepickerControl<any>, D>;
```

Comment:
can we replace `any` with `unknown` or `{}`?

Reply:
No, it ends up complaining that we can't pass in `unknown` to `ExtractDateTypeFromSelection`. We don't care about the value in this case anyway.

Reply:
ah ok makes sense